consensus/ibft/snapshot.go

```diff
@@ -191,87 +191,98 @@ func (i *Ibft) processHeaders(headers []*types.Header) error {
 			continue
 		}
 
-		// if we have a miner address, this might be a vote
-		if h.Miner == types.ZeroAddress {
-			continue
-		}
-
-		// the nonce selects the action
-		var authorize bool
-		if h.Nonce == nonceAuthVote {
-			authorize = true
-		} else if h.Nonce == nonceDropVote {
-			authorize = false
-		} else {
-			return fmt.Errorf("incorrect vote nonce")
-		}
+		// Disable voting temporarily for PoS
+		// if err := i.applyVote(h, proposer, snap); err != nil {
+		// 	return nil
+		// }
 
-		// validate the vote
-		if authorize {
-			// we can only authorize if they are not on the validators list
-			if snap.Set.Includes(h.Miner) {
-				continue
-			}
-		} else {
-			// we can only remove if they are part of the validators list
-			if !snap.Set.Includes(h.Miner) {
-				continue
+		if !snap.Equal(parentSnap) {
+			if err := saveSnap(h); err != nil {
+				return nil
```

Comment:
Isn't it strange that we return `nil` here if there is error, and not `err`?

Reply:
I agree with you. I fixed the code